scripts/assert-certificate-validity.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+CERTIFICATE=$1
+DATA=$2
+SIGNATURE=$3
+
+
+if [ -z "$CERTIFICATE" ]; then
+	echo "ERROR: Missing certificate parameter";
+	exit 1
+fi
+if [ -z "$DATA" ]; then
+	echo "ERROR: Missing data parameter";
+	exit 1
+fi
+if [ -z "$SIGNATURE" ]; then
+	echo "ERROR: Missing signature parameter";
+	exit 1
+fi
+
+openssl_verify()
+{
+	openssl smime -verify -in "$1" -inform der -content "$2" -CAfile "$3" > /dev/null 2>&1
+}
+
+create_fake_signature()
+{
+	openssl genpkey -algorithm RSA -out key.pem > /dev/null 2>&1
+	openssl dgst -sha256 -sign key.pem -out fake.sig "$DATA"
+	mv fake.sig "$TEMPSIG"
+	rm -f key.pem
+}
+
+corrupt_certificate()
+{
+	sed -i '10,$ d' "$TEMPCERT"
+}
+
+corrupt_data()
+{
+	echo "deadcafe" >> "$TEMPDATA"
+}
+
+set_tempfiles()
+{
+	cp "$CERTIFICATE" "$TEMPCERT"
+	cp "$DATA" "$TEMPDATA"
+	cp "$SIGNATURE" "$TEMPSIG"
+}
+
+clean_tempfiles()
+{
+	rm "$TEMPCERT" "$TEMPDATA" "$TEMPSIG"
+}
+# The tests follow this truth table to decide that the certificate and
+# signature are valid. In this table 1 refers to ERROR or INVALID, while 0
+# refers to SUCCESS or VALID, so the first line in the table means the
+# certificate, data, and signature are invalid and we expect an ERROR return
+#
+# +------+------+-----+-------+
+# | Cert | Data | Sig | Error |
```

Comment:
It wasn't entirely clear to me that the last column is the "result" column from the text.